packages/osd-optimizer/src/limits.ts

```diff
@@ -93,22 +93,37 @@ export function validateLimitsForAllBundles(log: ToolingLog, config: OptimizerCo
 export function updateBundleLimits(log: ToolingLog, config: OptimizerConfig) {
   const metrics = getMetrics(log, config);
 
-  const pageLoadAssetSize: NonNullable<Limits['pageLoadAssetSize']> = {};
+  let externalPlugins: { [id: string]: number } = {};
+  let osdPlugins: { [id: string]: number } = {};
 
-  for (const metric of metrics.sort((a, b) => a.id.localeCompare(b.id))) {
+  // Categorize all metrics
+  for (const metric of metrics) {
     if (metric.group === 'page load bundle size') {
       const existingLimit = config.limits.pageLoadAssetSize?.[metric.id];
-      pageLoadAssetSize[metric.id] =
+      const limitValue =
         existingLimit != null && existingLimit >= metric.value
           ? existingLimit
           : metric.value + DEFAULT_BUDGET;
+
+      if (metric.id.includes('Dashboards')) {
+        externalPlugins[metric.id] = limitValue;
+      } else {
+        osdPlugins[metric.id] = limitValue;
+      }
     }
   }
 
-  const newLimits: Limits = {
-    pageLoadAssetSize,
+  // Sort each category by their values
+  externalPlugins = Object.fromEntries(Object.entries(externalPlugins).sort((a, b) => b[1] - a[1]));
+  osdPlugins = Object.fromEntries(Object.entries(osdPlugins).sort((a, b) => b[1] - a[1]));
+
+  const newLimits: SortedLimits = {
+    pageLoadAssetSize: {
+      'External Plugins': externalPlugins,
```

Comment:
perhaps:
```
pageLoadAssetSize: {
  external: externalPlugins,
  internal: osdPlugins,
}
```
Since both categories are technically OpenSearch Dashboards Plugins